usr/share/rear/conf/default.conf

```diff
@@ -326,13 +326,24 @@ OUTPUT=ISO
 # Default cdrom size in MB (probably it is actually MiB?):
 CDROM_SIZE=20
 
-# The CHECK_CONFIG_FILES array lists files where changes
-# require the ReaR rescue/recovery system to be recreated.
+# The CHECK_CONFIG_FILES array lists files where changes require
+# that an up-to-date ReaR rescue/recovery system must be created anew.
 # Testing whether or not those files changed is implemented in the checklayout workflow
 # which exits with non-zero exit code when the disk layout or those files changed
 # (cf. https://github.com/rear/rear/issues/1134) but the checklayout workflow
 # does not automatically recreate the rescue/recovery system.
-CHECK_CONFIG_FILES=( '/etc/drbd/' '/etc/drbd.conf' '/etc/lvm/lvm.conf' '/etc/multipath.conf' '/etc/rear/' '/etc/udev/udev.conf' )
+# CHECK_CONFIG_FILES is also used in finalize/GNU/Linux/280_migrate_uuid_tags.sh
+# to check during "rear recover" if each UUID that was in at least one of the
+# config files in CHECK_CONFIG_FILES at the time when disklayout.conf was created
+# is also in at least one of the restored config files in CHECK_CONFIG_FILES
+# (i.e. to detect if restored config files do not match what was recreated).
+# Therefore all relevant config files that contain UUIDs should be specified here
+# (some files in /etc/ contain UUIDs but are not relevant like /etc/lvm/cache/.cache).
+# To find files in /etc/ that contain UUIDs on block devices one may use a command like
+#   for uuid in $( lsblk -no UUID,PTUUID,PARTUUID ) ; do find /etc -type f | xargs grep -il $uuid ; done | sort -u
+# which may show files in /etc/ that contain UUIDs which are not used in disklayout.conf
+# so one has to decide what is actually relevant for "rear recover":
+CHECK_CONFIG_FILES=( '/etc/fstab' '/etc/crypttab' '/etc/drbd/' '/etc/drbd.conf' '/etc/lvm/lvm.conf' '/etc/multipath.conf' '/etc/rear/' '/etc/udev/udev.conf' )
```

Comment:
@jsmeix A quick question (I have not yet read the entire change in detail, so sorry if it is answered in some of the comments). In `usr/share/rear/finalize/GNU/Linux/280_migrate_uuid_tags.sh`, former line 28, there is a long list of configuration file patterns to transform. Can't you just add all these files to `CHECK_CONFIG_FILES`? Wouldn't that itself solve the problem?

Reply:
I had noticed the long list of configuration file patterns in 280_migrate_uuid_tags.sh
and that it does not match what there is by default in CHECK_CONFIG_FILES
but CHECK_CONFIG_FILES gets appended by some scrips, cf.
```
# find usr/share/rear -type f | xargs grep -l 'CHECK_CONFIG_FILES+='

usr/share/rear/layout/save/default/450_check_network_files.sh
usr/share/rear/layout/save/default/450_check_bootloader_files.sh
usr/share/rear/layout/save/default/400_check_backup_special_files.sh
```

So a separated generic issue is to clean up our various lists
of configuration file patterns in various 'finalize' scripts
and align all of them with CHECK_CONFIG_FILES
but this is not something I will do in this pull request here.

In the end all 'finalize' scripts that modify restored files
should no longer have their own hardcoded file patterns.
Instead those file patterns should be in default.conf
likely as several separated arrays of file patterns
and CHECK_CONFIG_FILES would be the set union
of those separated arrays of file patterns.